From edb038b53cfa1a456557a27acd065ebaeb575b92 Mon Sep 17 00:00:00 2001
From: "Nalla, Pradeep" <pnalla@marvell.com>
Date: Fri, 15 Jan 2021 01:23:50 +0530
Subject: [PATCH 755/767] octeontx2-sdp/octeontx2-npa: Fix compiler warnings

commit 098acf6cea850c5905507e28bf68042828eeadba from
git@git.assembla.com:cavium/WindRiver.linux.git

These warnings are due to missing static declarations and unused
variables.

Signed-off-by: Nalla, Pradeep <pnalla@marvell.com>
Change-Id: I2daf1ccae930bafce250aae83af3a2c75146a029
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/43662
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49039
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-sdp/sdp.c | 16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-sdp/sdp.c b/drivers/soc/marvell/octeontx2-sdp/sdp.c
index 4319abfb2d7f..9398edf7cef2 100644
--- a/drivers/soc/marvell/octeontx2-sdp/sdp.c
+++ b/drivers/soc/marvell/octeontx2-sdp/sdp.c
@@ -313,16 +313,6 @@ reply_free_rsrc_cnt(struct sdp_dev *sdp, struct rvu_vf *vf,
 	return 0;
 }
 
-static int
-check_attach_rsrcs_req(struct sdp_dev *sdp, struct rvu_vf *vf,
-		       struct mbox_msghdr *req, int size)
-{
-	struct rsrc_attach *rsrc_req;
-
-	rsrc_req = (struct rsrc_attach *)req;
-	return forward_to_mbox(sdp, &sdp->afpf_mbox, 0, req, size, "AF");
-}
-
 static int
 handle_vf_req(struct sdp_dev *sdp, struct rvu_vf *vf, struct mbox_msghdr *req,
 	      int size)
@@ -364,7 +354,7 @@ handle_vf_req(struct sdp_dev *sdp, struct rvu_vf *vf, struct mbox_msghdr *req,
 				req->ver, OTX2_MBOX_VERSION);
 			return -EINVAL;
 		}
-		err = check_attach_rsrcs_req(sdp, vf, req, size);
+		err = forward_to_mbox(sdp, &sdp->afpf_mbox, 0, req, size, "AF");
 		break;
 	default:
 		err = forward_to_mbox(sdp, &sdp->afpf_mbox, 0, req, size, "AF");
@@ -1004,7 +994,7 @@ static struct attribute_group sdp_ring_attr_group = {
 	.attrs = sdp_ring_attrs,
 };
 
-int sdp_sysfs_init(struct device *dev)
+static int sdp_sysfs_init(struct device *dev)
 {
 	int ret;
 
@@ -1017,7 +1007,7 @@ int sdp_sysfs_init(struct device *dev)
 	return 0;
 }
 
-void sdp_sysfs_remove(struct device *dev)
+static void sdp_sysfs_remove(struct device *dev)
 {
 	sysfs_remove_group(&dev->kobj, &sdp_ring_attr_group);
 }
-- 
2.31.1

